packages/compiler/server/serverlib.ts

```diff
@@ -618,9 +622,55 @@ export function createServer(host: ServerHost): Server {
   async function addImportCompletion(
     program: Program,
     document: TextDocument,
-    completions: CompletionList
+    completions: CompletionList,
+    node: StringLiteralNode
+  ) {
+    if (node.value.startsWith("./") || node.value.startsWith("../")) {
+      await addRelativePathCompletion(program, document, completions, node);
+    } else if (!node.value.startsWith(".")) {
+      await addLibraryImportCompletion(program, document, completions);
+    }
+  }
+
+  async function addRelativePathCompletion(
+    program: Program,
+    document: TextDocument,
+    completions: CompletionList,
+    node: StringLiteralNode
   ) {
-    await addLibraryImportCompletion(program, document, completions);
+    const documentPath = getPath(document);
+    const documentFile = getBaseFileName(documentPath);
+    const documentDir = getDirectoryPath(documentPath);
+    if (documentDir.indexOf("node_modules") !== -1) {
+      return;
+    }
+    const nodevalueDir = getDirectoryPath(node.value);
```

Comment:
I think this is where we end up with a bug where you have import "./folder/" and it offers "./folder/folder"